chandra_aca/aca_image.py

```diff
@@ -309,6 +311,257 @@ def col0(self):
     def col0(self, value):
         self.meta['IMGCOL0'] = np.int64(value)
 
+    @classmethod
+    def _read_flicker_cdfs(cls):
+        """Read flickering pixel model cumulative distribution functions
+        and associated metadata.  Set up class variables accordingly.
+
+        The flicker_cdf file here was created using:
+        /proj/sot/ska/www/ASPECT/ipynb/chandra_aca/flickering-pixel-model.ipynb
+
+        """
+        from astropy.io import fits
+
+        filename = Path(__file__).parent / 'data' / 'flicker_cdf.fits.gz'
+        with fits.open(filename) as hdus:
+            hdu = hdus[0]
+            hdr = hdu.header
+
+            # Get the main data, which is an n_cdf * n_cdf_x array.  Each row corresponds
+            # to the CDF for a particular bin range in e-/sec, e.g. 200 to 300 e-/sec.
+            # CDF will go from 0.0 to 1.0
+            cls.flicker_cdfs = hdu.data.astype(np.float64)
+
+            # CDF_x is the x-value of the distribution, namely the log-amplitude change
+            # in pixel value due to a flicker event.
+            cls.flicker_cdf_x = np.linspace(hdr['cdf_x0'], hdr['cdf_x1'], hdr['n_cdf_x'])
+
+            # CDF bin range (e-/sec) for each for in flicker_cdfs.
+            cdf_bins = []
+            for ii in range(hdr['n_bin']):
+                cdf_bins.append(hdr[f'cdf_bin{ii}'])
+            cls.flicker_cdf_bins = np.array(cdf_bins)
+
+    def flicker_init(self, flicker_mean_time=10000, flicker_scale=1.0, seed=None):
+        """Initialize instance variables to allow for flickering pixel updates.
+
+        The ``flicker_scale`` can be interpreted as follows: if the pixel
+        was going to flicker by a multiplicative factor of (1 + x), now
+        make it flicker by (1 + x * flicker_scale).  This applies for flickers
+        that increase the amplitude.  For flickers that make the value smaller,
+        then it would be 1 / (1 + x) => 1 / (1 + x * flicker_scale).
+
+        The flicker_cdf file here was created using:
+        /proj/sot/ska/www/ASPECT/ipynb/chandra_aca/flickering-pixel-model.ipynb
+
+        Examples and performance details at:
+        /proj/sot/ska/www/ASPECT/ipynb/chandra_aca/flickering-implementation.ipynb
+
+        The model was reviewed and approved at SS&AWG on 2019-05-22.
+
+        :param flicker_mean_time: mean flickering time (sec, default=10000)
+        :param flicker_scale: multiplicative factor beyond model default for
+               flickering amplitude (default=1.0)
+        :param seed: random seed for reproducibility (default=None => no seed)
+        """
+        if not hasattr(self, 'flicker_cdf_bins'):
+            self._read_flicker_cdfs()
+
+        self.flicker_mean_time = flicker_mean_time
+        self.flicker_scale = flicker_scale
+        self.test_idx = 1 if seed == -1 else 0
+
+        if seed is not None and seed != -1:
+            np.random.seed(seed)
+            _numba_random_seed(seed)
```

Comment:
It has been a while since I've looked at random number reproducibility.  Does this need to set both seed and state or is seed sufficient?

Reply:
I can't say definitively except that:
- The test is reproducibly passing, so setting only `seed` apparently results in the same random sequence.
- Having the numpy.seed() not do the whole job would be pretty silly, that's the whole point of advertising a function for setting a seed to get a reproducible sequence.

Reply:
Gotcha.  I thought I remembered something about setting the seed and starting the number generator from "clean" starting state was reproducible, but if the the number generator was in  a "used" state, setting the seed again does not quite put the number generator in the same state?   Eh.  If we notice some weirdness down the line we can obviously come back to this.

Reply:
You need to erase that memory and replace with the bug-fixed one:
```
In [24]: np.random.seed(1)
In [25]: x = np.random.uniform(size=1000)
In [26]: state1 = np.random.get_state()
In [27]: np.random.seed(1)
In [28]: y = np.random.uniform(size=1000)
In [29]: state2 = np.random.get_state()

In [30]: np.all(x == y)
Out[30]: True

In [31]: [np.all(s1 == s2) for s1, s2 in zip(state1, state2)]
Out[31]: [True, True, True, True, True]
```

Reply:
If the seed is sufficient (great!) would there any value in saving the seed value in the ACAImage object?

Reply:
No.